src/connection/builders/builder_native.rs

```diff
@@ -0,0 +1,301 @@
+        self
+    }
+
+    /// SQL Dialect. Default: 3
+    pub fn dialect(&mut self, dialect: Dialect) -> &mut Self {
+        self.conn_conf.dialect = dialect;
+        self
+    }
+
+    /// SQL Dialect. Default: 3
+    pub fn charset(&mut self, charset: Charset) -> &mut Self {
+        self.charset = charset;
+        self
+    }
+
+    /// Statement cache size. Default: 20
+    pub fn stmt_cache_size(&mut self, stmt_cache_size: usize) -> &mut Self {
+        self.conn_conf.stmt_cache_size = stmt_cache_size;
+        self
+    }
+}
+
+impl<A, B> NativeConnectionBuilder<A, B> {
+    //never export this. It would allow users to bypass the type safety.
+    fn safe_transmute<X, Y>(self) -> NativeConnectionBuilder<X, Y> {
+        NativeConnectionBuilder {
+            _marker_linkage: PhantomData,
+            _marker_conntype: PhantomData,
+            conn_conf: self.conn_conf,
+            charset: self.charset,
+            lib_path: self.lib_path,
+        }
+    }
+}
+
+impl Default for NativeConnectionBuilder<LinkageNotConfigured, ConnTypeNotConfigured> {
+    fn default() -> Self {
+        let mut self_result = Self {
+            _marker_linkage: PhantomData,
+            _marker_conntype: PhantomData,
+            conn_conf: Default::default(),
+            charset: charset::UTF_8,
+            lib_path: None,
+        };
+
+        self_result.conn_conf.dialect = Dialect::D3;
+        self_result.conn_conf.stmt_cache_size = 20;
+        self_result.conn_conf.attachment_conf.remote = None;
+        self_result.conn_conf.attachment_conf.user = "SYSDBA".to_string();
+        self_result.conn_conf.attachment_conf.db_name = "test.fdb".to_string();
+
+        self_result
+    }
+}
+
+//can only use these methods on a remote builder
+impl<A> NativeConnectionBuilder<A, Remote> {
+    //private helper accessor for the Option<RemoteConfig> buried inside
+    //the configuration
+    fn get_initialized_remote(&mut self) -> &mut RemoteConfig {
+        self.conn_conf
+            .attachment_conf
+            .remote
+            .get_or_insert(Default::default())
+    }
+
+    /// Hostname or IP address of the server. Default: localhost
+    pub fn host<S: Into<String>>(&mut self, host: S) -> &mut Self {
+        self.get_initialized_remote().host = host.into();
+        self
+    }
+
+    /// TCP Port of the server. Default: 3050
+    pub fn port(&mut self, port: u16) -> &mut Self {
+        self.get_initialized_remote().port = port;
+        self
+    }
+
+    /// Password. Default: masterkey
+    pub fn pass<S: Into<String>>(&mut self, pass: S) -> &mut Self {
+        self.get_initialized_remote().pass = pass.into();
+        self
+    }
+}
+
+impl<A> NativeConnectionBuilder<A, ConnTypeNotConfigured> {
+    /// Configure the native client for remote connections.
+    /// This will allow configuration via the 'host', 'port' and 'pass' methods.
+    pub fn as_remote(mut self) -> NativeConnectionBuilder<A, Remote> {
```

Comment:
Clippy warning.
Consider changing to `into_remote` or suppressing the warning using `#[allow(clippy::wrong_self_convention)]`.